ces_admin/management/commands/hash_identities.py

```diff
@@ -0,0 +1,34 @@
+import datetime
+import hashlib
+
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+
+from rapidsms.models import Connection
+from decisiontree.models import Session
+
+class Command(BaseCommand):
+    help = 'Hashes the user identities, i.e., phone numbers'
+
+    def handle(self, *args, **options):
+        five_minutes_ago = datetime.datetime.now() - datetime.timedelta(minutes=5)
+        one_day_ago = datetime.datetime.now() - datetime.timedelta(days=1)
+        # Find connections from sessions that do not have a state_id, i.e., the user completed the survey or canceled their session
+        # Add a five minute boundary, in the rare event that Rapidsms or Twilio 
+        # Exclude sessions that have already been hased or are websessions. 
+        filter_for_closed = Q(session__state_id__isnull=True, session__last_modified__lte=five_minutes_ago)
+        filter_for_abandoned = Q(session__last_modified__lte=one_day_ago)
+
+        connections_from_closed_sessions = Connection.objects \
+                                                     .filter(filter_for_closed | filter_for_abandoned) \
+                                                     .filter(identity__startswith='+1') \
+                                                     .distinct()
+
+        for connection in connections_from_closed_sessions:
+            identity = connection.identity
+            hashed_identity = hashlib.sha256(identity.encode()).hexdigest()
```

Comment:
I'd recommend using django's in-built hashers https://docs.djangoproject.com/en/2.1/topics/auth/passwords/

Reply:
all right, I used `make_password` from Django [(which can be implemented independent of the User model)](https://docs.djangoproject.com/en/1.11/topics/auth/passwords/#module-django.contrib.auth.hashers). 

I think it makes sense - however, on the Rapidsms side of things, I have to iterate over all connections and run `check_password`, to find previously hashed identities:
https://github.com/datamade/rapidsms/pull/3/files#diff-632733e9d1d3abf4062a87fc96a1a840R101 Could this become inefficient if we have, say, thousands of connections in the database?